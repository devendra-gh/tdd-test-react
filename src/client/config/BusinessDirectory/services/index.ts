import getMapNatureData from './getMapNatureData';
import getMapActivityData from './getMapActivityData';
import getToken from './getToken';

/* istanbul ignore file */

export { getMapActivityData, getMapNatureData, getToken };
