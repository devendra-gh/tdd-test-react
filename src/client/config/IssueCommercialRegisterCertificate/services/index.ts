import { IVariables } from '@tamm/app-composer';
import fetch from 'client/services/fetch';
import baseUrl from 'client/utils/baseUrl';
import { PATH_ERROR } from '../routes';

export const fetchLicences = async () => {
  return fetch('/pub/proxy/listTradeLicensesV3', 'POST', {}).then(
    (response: IVariables) => {
      let licences = [];
      const {
        responseStatus: { statusCode },
      } = response.data;
      if (statusCode === 200) {
        licences = response.data.TradeLicensesList.contents;
      }
      return licences;
    },
  );
};

export const issueCertificate = async ({ cnNumber }: IVariables) => {
  return fetch('/pub/proxy/issueLetter', 'POST', {
    cnNumber,
    letterType: 'license',
  }).then(({ success, data }) => {
    if (success) {
      return data;
    }
    return null;
  });
};

export const downloadBusinessCertificate = async ({
  instanceId,
  certificateName,
  type,
}: IVariables) => {
  window.open(
    `${baseUrl}/api/download/businessCertificateGenericAuthADU?instanceId=${instanceId}&type=${type}&certificateName=${certificateName}&mobileDownloadable=pdf&mobileFileName=Certificate`,
  );
};

export const errorBoundary = (props: IVariables, error: any) => {
  // console.log('error fetching user licenses', error); // eslint-disable-line
  props.history.push(PATH_ERROR);
};
