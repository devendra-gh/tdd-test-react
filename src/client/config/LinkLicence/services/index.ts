import fetch from 'client/services/fetch';

// TODO: get the details about the licence
const industrialLicenceDetails = (licenseNo: string) => {
  return fetch('/pub/proxy/getLicenceDetailsV3', 'POST', {
    licenseNo: licenseNo || '',
  })
    .then(response => {
      const details = {
        code: response.success ? '204' : '500',
        message: response.data.message,
        traceId: response.data.traceId,
      };
      if (response.success && response.data && response.data.result) {
        if (response.data.result.constructor === Array) {
          const data = response.data.result[0];
          Object.assign(details, {
            code: response.data.code,
            licenceNumber: data.licenseNo,
            licenceTypeEn: data.clasification_en,
            licenceTypeAr: data.clasification_ar,
            tradeNameEn: data.businessNameEng,
            tradeNameAr: data.businessNameArb,
          });
        }
        return details;
      }
      throw details;
    })
    .catch(err => {
      throw err;
    });
};

// TODO: get link licence status
const linkUserLicence = async (
  licenseNo: string,
  spuuid: string,
  userEmail: string,
  userMobile: string,
) => {
  try {
    return await fetch('/pub/proxy/linkLicense', 'POST', {
      licenseNumber: licenseNo || '',
      SPUUID: spuuid || '',
      userEmail,
      userMobile,
    });
  } catch (error) {
    throw error;
  }
};

export const services: { [key: string]: Function } = {
  industrialLicenceDetails,
  linkUserLicence,
};
