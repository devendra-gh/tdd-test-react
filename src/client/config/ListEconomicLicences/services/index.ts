import { IVariables } from '@tamm/app-composer';
import fetch from 'client/services/fetch';
import baseUrl from 'client/utils/baseUrl';
import { PATH_ERROR } from '../routes';

export const issueCertificate = async ({ cnNumber }: IVariables) => {
  return fetch('/pub/proxy/issueLetter', 'POST', {
    cnNumber,
    letterType: 'haslicense',
  }).then(({ success, data }) => {
    if (success) {
      return data;
    }
    return null;
  });
};

export const downloadBusinessCertificate = async ({
  instanceId,
  certificateName,
  type,
}: IVariables) => {
  window.open(
    `${baseUrl}/api/download/businessCertificateGenericAuthADU?instanceId=${instanceId}&type=${type}&certificateName=${certificateName}&mobileDownloadable=pdf&mobileFileName=Certificate`,
  );
};

export const errorBoundary = (props: IVariables, error: any) => {
  props.history.push(PATH_ERROR);
};
