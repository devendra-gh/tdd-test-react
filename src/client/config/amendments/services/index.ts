import getLicenceDetails from './getLicenceDetails';
import getActivities from './getActivities';
import getRelatedRecords from './getRelatedRecords';
import checkApplicationStatus from './checkApplicationStatus';
import updateReduxState from './updateReduxState';
import getReduxState from './getReduxState';

export {
  getLicenceDetails,
  getActivities,
  getRelatedRecords,
  checkApplicationStatus,
  updateReduxState,
  getReduxState,
};
