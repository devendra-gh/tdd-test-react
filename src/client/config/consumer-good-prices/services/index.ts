import fetchGoodsFromSearchText from './goodsSearch';
import fetchGoodsDetail from './getItem';
import fetchCategories from './categories';
import fetchGoodsFromCategory from './goodsCategories';

export {
  fetchGoodsFromSearchText,
  fetchGoodsDetail,
  fetchCategories,
  fetchGoodsFromCategory,
};
