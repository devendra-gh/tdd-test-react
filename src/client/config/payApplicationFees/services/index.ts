import checkApplicationStatus from './checkApplicationStatus';
import fetchLicenceDetails from './fetchLicenceDetails';
import applicationFees from './applicationFees';
import getVariables from './bpm';

export default {
  checkApplicationStatus,
  fetchLicenceDetails,
  applicationFees,
  getVariables,
};
