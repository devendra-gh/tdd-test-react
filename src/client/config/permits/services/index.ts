import fetch from 'client/services/fetch';
import { permitType } from '../utils/constants';

export const fetchRequirementsInquiry = async (serviceName: string) => {
  return fetch('/api/proxy/permitRequirementsInquiry', 'POST', {
    permitType: permitType[serviceName].id,
  });
};
