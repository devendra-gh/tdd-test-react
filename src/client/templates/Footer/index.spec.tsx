import React from 'react';
import { shallow } from 'enzyme';
import { render, cleanup, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import { hasFeatureFlag } from 'client/services/featureFlag';
import { FooterTemplate } from './index';

jest.mock('client/services/featureFlag');

jest.mock('client/utils/appData', () => ({
  getCMSData: () => ({
    journeyInfo: {
      en: {
        footerInfo: {
          copyrightMessage: 'copyrightMessage',
          emergencyNumbers: [
            {
              name: 'Police',
              number: '999',
            },
          ],
          viewAllEmergencyNumbers: 'viewAllEmergencyNumbers',
          translateLabel: 'translateLabel',
          fontSizeLabel: 'fontSizeLabel',
          colorLabel: 'colorLabel',
          logos: [
            {
              image: '/-/media/logo.png',
            },
          ],
          downloadApps: [
            {
              appImage: '/-/media/apple-store.png',
              appLink: 'http://www.google.com',
              appcaption: 'Apple Store',
            },
          ],
          copyrights: [
            {
              copyrightName: 'Privacy Policy',
              copyrightLink: '/en/privacy-statement',
            },
          ],
          socialLinks: [
            {
              socialSiteImage: '/-/media/social-media-twitter.svg',
              socialSiteLink: 'http://www.google.com',
              socialSiteCaption: 'Twitter',
            },
          ],
        },
      },
    },
  }),
}));

describe('client/templates/Footer', () => {
  let props: any;

  beforeEach(() => {
    props = {
      match: {},
      location: {},
      locale: 'en',
      actions: {
        hero: {
          update: jest.fn(),
        },
        breadcrumbs: {
          update: jest.fn(),
        },
      },
    };
  });

  afterEach(cleanup);

  it('should render with props', async () => {
    window.matchMedia =
      window.matchMedia ||
      function () {
        return {
          matches: false,
          addListener() {},
          removeListener() {},
        };
      };

    await act(async () => {
      const { getByText } = render(
        <MemoryRouter>
          <ViewportProvider>
            <Viewport sm md lg xl>
              <FooterTemplate {...props} />
            </Viewport>
          </ViewportProvider>
        </MemoryRouter>,
      );

      getByText('copyrightMessage');
    });
  });

  it('should render with feature flag', async () => {
    (hasFeatureFlag as jest.Mock).mockReturnValue(true);
    const component = shallow(<FooterTemplate />);

    expect(component).toMatchSnapshot();
  });
});
