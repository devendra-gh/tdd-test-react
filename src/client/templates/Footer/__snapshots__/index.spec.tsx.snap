// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/templates/Footer should render with feature flag 1`] = `
<Fragment>
  <Feedback />
  <BugReport />
  <LocalizedComponent(CookieTray)
    cookieTrayStorageKey="ui-lib-cookey-key"
    i18n={[Function]}
    locale="en"
    onClick={[Function]}
    onClose={[Function]}
    privacyLink=""
    privacyLinkTarget="_blank"
  />
  <LocalizedComponent(Footer)
    appLinks={Array []}
    bottomRightLinks={Array []}
    colorLabel="footer.theme_switcher.label"
    copyright={
      Object {
        "label": undefined,
      }
    }
    defaultThemeValue="footer.theme_switcher.default"
    emergencyItems={Array []}
    emergencyViewMoreLinkTarget="_blank"
    emiratesLogo={
      Object {
        "iconPath": undefined,
      }
    }
    fontSizeLabel="footer.text_size"
    i18n={[Function]}
    locale="en"
    logUuid=""
    onTextSizeChange={[Function]}
    onThemeChange={[Function]}
    socialLinks={Array []}
    translateLabel="footer.language.label"
  />
</Fragment>
`;
